cmd/soroban-rpc/internal/methods/get_events.go

```diff
@@ -392,35 +409,60 @@ func eventInfoForEvent(event xdr.DiagnosticEvent, cursor events.Cursor, ledgerCl
 		return EventInfo{}, fmt.Errorf("unknown XDR ContractEventType type: %d", event.Event.Type)
 	}
 
-	// base64-xdr encode the topic
-	topic := make([]string, 0, 4)
-	for _, segment := range v0.Topics {
-		seg, err := xdr.MarshalBase64(segment)
-		if err != nil {
-			return EventInfo{}, err
-		}
-		topic = append(topic, seg)
-	}
-
-	// base64-xdr encode the data
-	data, err := xdr.MarshalBase64(v0.Data)
-	if err != nil {
-		return EventInfo{}, err
-	}
-
 	info := EventInfo{
 		EventType:                eventType,
 		Ledger:                   int32(cursor.Ledger),
 		LedgerClosedAt:           ledgerClosedAt,
 		ID:                       cursor.String(),
 		PagingToken:              cursor.String(),
-		Topic:                    topic,
-		Value:                    data,
 		InSuccessfulContractCall: event.InSuccessfulContractCall,
 		TransactionHash:          txHash,
 	}
+
+	switch format {
+	case xdr2json.FormatJSON:
+		// json encode the topic
+		topics := make([]json.RawMessage, 0, maxTopicCount)
+		for _, topic := range v0.Topics {
+			topic, err := xdr2json.ConvertInterface(topic)
+			if err != nil {
+				return EventInfo{}, err
+			}
+			topics = append(topics, topic)
+		}
+		info.TopicJSON = topics
+
+		dataJs, err := xdr2json.ConvertInterface(v0.Data)
```

Comment:
nit: since the topics variable defined above does not have the Js suffix, for the sake of consistency, I think we should rename this variable to `data`

Reply:
sure, fair enough! refactored a bit to not use temporary variables and have consistent naming :+1: